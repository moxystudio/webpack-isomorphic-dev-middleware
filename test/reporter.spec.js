'use strict';

const express = require('express');
const request = require('supertest');
const webpackIsomorphicDevMiddleware = require('../');
const createCompiler = require('./util/createCompiler');
const createWritter = require('./util/createWritter');
const touchFile = require('./util/touchFile');
const configClientBasic = require('./configs/client-basic');
const configServerBasic = require('./configs/server-basic');

afterEach(() => jest.restoreAllMocks());
afterEach(() => createCompiler.teardown());

it('should report stats only once by default', async () => {
    const app = express();
    const compiler = createCompiler(configClientBasic, configServerBasic);
    const writter = createWritter();

    app.use(webpackIsomorphicDevMiddleware(compiler, {
        report: { write: writter },
    }));

    await request(app)
    .get('/client.js')
    .expect(200);

    await new Promise((resolve) => {
        compiler.once('begin', resolve);
        // Need to add a little delay otherwise webpack won't pick it up..
        // This happens because the file is being written while chokidar is not yet ready (`ready` event not yet emitted)
        setTimeout(() => touchFile(configServerBasic.entry), 200);
    });

    await request(app)
    .get('/client.js')
    .expect(200);

    expect(writter.getReportOutput()).toMatchSnapshot();
});

it('should not report anything if options.report is false', async () => {
    const app = express();
    const compiler = createCompiler(configClientBasic, configServerBasic);

    jest.spyOn(process.stderr, 'write');
    app.use(webpackIsomorphicDevMiddleware(compiler, {
        report: false,
    }));

    await request(app)
    .get('/client.js')
    .expect(200);

    expect(process.stderr.write).toHaveBeenCalledTimes(0);
});
