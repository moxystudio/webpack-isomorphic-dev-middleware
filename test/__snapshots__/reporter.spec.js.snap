// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should report stats only once by default 1`] = `
"● Compiling...
✔ Compilation succeeded (xxx)

     CLIENT                             
    ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
    Asset Size Chunks Chunk Names
    client.js  xxx

     SERVER                             
    ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
    Asset Size Chunks Chunk Names
    server.js  xxx

● Compiling...
✔ Compilation succeeded (xxx)
"
`;
